biogrid/libs/biogrid-simulator/src/lib/biobattery/biobattery-class.ts

```diff
@@ -24,25 +42,27 @@ export class BioBattery implements Battery {
    * @param x Distance from the left edge of the town
    * @param y Distance from the top edge of the town
    */
-  constructor(
-    x: Distance,
-    y: Distance,
-    gridItemName: string,
-    gridItemResistance: number,
-    energyInJoules: Energy = SMALL_BATTERY.DEFAULT_START_ENERGY,
-    maxCapacity: Energy = SMALL_BATTERY.MAX_CAPACITY
-  ) {
-    this.relativePosition = { x, y };
-    if (!this.validateInputs(energyInJoules, maxCapacity)) {
+  constructor(params: BatteryParams) {
+    const paramsWithDefault = { ...BATTERY_PARAMS_DEFAULT, ...params };
```

Comment:
I don't think we want default params here. TS will check that all the necessary params are present, and we want to have the user set all these values instead of allowing them to default to an empty string id or other default values
@cwlarson-google feel free to correct me if I'm wrong here - I'm going off a comment you made on a separate PR

Reply:
If you have optional params, the "standard" way that I've seen (could be something my team made up, could be a real standard) is to set the default when declaring the class property and then conditionally overwriting it if a value is passed in:

```
export class BioBattery implements Battery {
  private energyInJoules: Energy = 100;    // Defaults to 100 Energy

  constructor(params: BatteryParams) {
    if (params.energyInJoules) {
      this.energyInJoules = params.energyInJoules;
    }
  }
}
```

Since you need to run things through a validator, this might make things more complicated?

At the end of the day, I think consistency is more important than anything else.  If this is how the whole app is approaching defaults, then fine

Reply:
Ok sure, it should be standard now